swiftide-agents/src/tasks/running_agent.rs

```diff
@@ -0,0 +1,40 @@
+//! A running agent wraps an agent and its name, and is cheap to clone with inner mutability.
+//!
+//! This allows tasks to keep track of running agents without needing to lock the list of agents
+//! itself.
+//!
+//! This allows
+
+use std::{ops::Deref, sync::Arc};
+
+use crate::Agent;
+#[derive(Clone, Debug)]
+pub(crate) struct RunningAgent(Arc<tokio::sync::Mutex<Agent>>, Arc<String>);
+
+impl RunningAgent {
```

Comment:
Maybe name it a `TaskAgent` instead. Technically it's not 'running', it's just wrapped with inner mutability and a ref for easy access.